projects/ion/src/lib/notification/service/notification.service.ts

```diff
@@ -0,0 +1,204 @@
+import { StatusType } from './../../core/types/status';
+import { NotificationConfigOptions } from './../../core/types/notification';
+import { IonNotificationComponent } from './../component/notification.component';
+import {
+  Injectable,
+  ComponentRef,
+  Inject,
+  ComponentFactoryResolver,
+  ApplicationRef,
+  Injector,
+} from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+import { Subject } from 'rxjs';
+import { SafeAny } from './../../utils/safe-any';
+import { IonNotificationContainerComponent } from './notification.container.component';
+
+enum NOTIFICATION_TYPES {
+  success = 'success',
+  info = 'info',
+  warning = 'warning',
+  negative = 'negative',
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class IonNotificationService {
+  private notificationContainerComponentRef: ComponentRef<IonNotificationContainerComponent>;
+  private componentSubscriber!: Subject<SafeAny>;
+
+  constructor(
+    @Inject(DOCUMENT) private document: SafeAny,
+    private componentFactoryResolver: ComponentFactoryResolver,
+    private appRef: ApplicationRef,
+    private injector: Injector
+  ) {}
+
+  public success(
+    title: string,
+    message: string,
+    options?: NotificationConfigOptions,
+    closeEventCall?: () => void
+  ): void {
+    if (!this.notificationContainerComponentRef)
+      this.createNotificationContainer();
+
+    const notification = this.createNotificationInstance();
+
+    this.configNotification(notification.instance, title, message, options);
+
+    this.instanceNotification(notification);
+
+    if (closeEventCall) {
+      this.addCloseEventEmitter(notification, closeEventCall);
+    }
+  }
+
+  public info(
+    title: string,
+    message: string,
+    options?: NotificationConfigOptions,
+    closeEventCall?: () => void
+  ): void {
+    if (!this.notificationContainerComponentRef)
+      this.createNotificationContainer();
+
+    const notification = this.createNotificationInstance();
+
+    this.configNotification(
+      notification.instance,
+      title,
+      message,
+      options,
+      NOTIFICATION_TYPES.info
+    );
+
+    this.instanceNotification(notification);
+
+    if (closeEventCall) {
+      this.addCloseEventEmitter(notification, closeEventCall);
+    }
+  }
+
+  public warning(
+    title: string,
+    message: string,
+    options?: NotificationConfigOptions,
+    closeEventCall?: () => void
+  ): void {
+    if (!this.notificationContainerComponentRef)
+      this.createNotificationContainer();
+
+    const notification = this.createNotificationInstance();
+
+    this.configNotification(
+      notification.instance,
+      title,
+      message,
+      options,
+      NOTIFICATION_TYPES.warning
+    );
+
+    this.instanceNotification(notification);
+
+    if (closeEventCall) {
+      this.addCloseEventEmitter(notification, closeEventCall);
+    }
```

Comment:
I believe that you can create a method called showNotification and put this code and call method in all methods by type. (sucess, warning) like this:

```suggestion
    this.showNotification(title, message, options, closeEventCall, NOTIFICATION_TYPES.warning)
```